syntax = "proto3";

package api;
import "diff.proto";

// Route /tasks/get, /tasks/update
message Task {
	option (diffable) = true;

	int64 id = 1;

	optional int64 task_type = 2;
	optional int64 parent = 3;
	optional string text = 4;
	optional int32 answer_rows = 5;
	optional int32 answer_cols = 6;
	optional bytes answer = 7;
	optional string tag = 9;

	message Attachment {
		int64 id = 1;

		optional string filename = 2;
		optional string mime_type = 3;
		bytes contents = 5 [(out) = true];
		string hash = 6 [(in) = true];
		optional bool shown_to_user = 7;
		optional bool deleted = 8;
	}
	repeated Attachment attachments = 8;
}
